From a325b018ef821b15a8ad6a6fdf38e4fdff3ea26c Mon Sep 17 00:00:00 2001
From: Philipp Rosenberger <p.rosenberger@kunbus.com>
Date: Fri, 26 Mar 2021 10:34:04 +0100
Subject: [PATCH] Move the log file from /usr/share to /var/log

Logging to /usr/share is problematic andby the FSH logs should go to
/var/log. Otherwise it is hard to create a read-only rfs. Also /var
might be handaled differently to avoid flash wear out.

Signed-off-by: Philipp Rosenberger <p.rosenberger@kunbus.com>
---
 bin/noderedrevpinodes-server.service | 3 ++-
 revpi-server.py                      | 2 +-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/bin/noderedrevpinodes-server.service b/bin/noderedrevpinodes-server.service
index 645eb37..6df0634 100644
--- a/bin/noderedrevpinodes-server.service
+++ b/bin/noderedrevpinodes-server.service
@@ -5,7 +5,8 @@ Documentation="https://github.com/erminas/noderedrevpinodes-server"
 
 [Service]
 Type=simple
-ExecStart=/bin/sh -c 'exec /usr/bin/python3 -u revpi-server.py 2>> revpi-server.log'
+StandardOutput=append:%L/revpi-server.log
+ExecStart=/usr/bin/python3 -u revpi-server.py
 WorkingDirectory=/usr/share/noderedrevpinodes-server/
 Restart=on-abort
 
diff --git a/revpi-server.py b/revpi-server.py
index 1dd837f..2c59374 100644
--- a/revpi-server.py
+++ b/revpi-server.py
@@ -43,7 +43,7 @@ root = logging.getLogger()
 if root.handlers:
     for handler in root.handlers:
         root.removeHandler(handler)
-logging.basicConfig(handlers=[RotatingFileHandler('revpi-server.log', maxBytes=100000000, backupCount=5)],
+logging.basicConfig(handlers=[RotatingFileHandler('/var/log/revpi-server.log', maxBytes=100000000, backupCount=5)],
                     level=logging.INFO,
                     format='%(asctime)s %(name)-12s: %(levelname)-8s %(message)s')
 
-- 
2.29.2

